src/components/accessibility/screen_reader_only/screen_reader_only.tsx

```diff
@@ -6,37 +6,46 @@
  * Side Public License, v 1.
  */
 
-import { cloneElement, ReactElement, FunctionComponent } from 'react';
+import { ReactElement, FunctionComponent } from 'react';
 import classNames from 'classnames';
 
+import { cloneElementWithCss } from '../../../services/theme/clone_element';
+import { euiScreenReaderOnlyStyles } from './screen_reader_only.styles';
+
 export interface EuiScreenReaderOnlyProps {
   /**
    * ReactElement to render as this component's content
    */
-  children: ReactElement<any>;
+  children: ReactElement;
 
   /**
    * For keyboard navigation, force content to display visually upon focus.
    */
   showOnFocus?: boolean;
+  className?: string;
 }
 
 export const EuiScreenReaderOnly: FunctionComponent<EuiScreenReaderOnlyProps> = ({
   children,
+  className,
   showOnFocus,
+  ...rest
 }) => {
   const classes = classNames(
     {
       euiScreenReaderOnly: !showOnFocus,
-      'euiScreenReaderOnly--showOnFocus': showOnFocus,
     },
+    className,
     children.props.className
   );
```

Comment:
I think you should remove the `euiScreenReaderOnly: !showOnFocus` line, so the end output should be something like this:

```suggestion
  const classes = classNames(
    'euiScreenReaderOnly',
    className,
    children.props.className
  );
```

Reply:
We need the `showOnFocus` logic here. Otherwise the `.euiScreenReaderOnly` global utility class will override the show-on-focus classes
```
&:not(:focus):not(:active) {
  ${euiScreenReaderOnly()}
}
```

and never allow for showing on focus.

Reply:
Or alternatively, can we separate the component className hook and the utility className somehow? e.g.,

- `.euiScreenReaderOnly` <- used by component as empty hook? (would need to update all direct usages in Kibana..)
- `.euiScreenReaderOnlyUtil` <-- utility? or maybe `.u-euiScreenReaderOnly` to take a leaf from other css libraries?

Honestly I'm not a super huge fan of either approach, and thinking about it more it feels really weird to apply both the global utility and the Emotion class. Since this component now allows passing in a custom `className` for consumer targeting purposes, can we instead opt to remove the global utility className from this component entirely since it isn't needed?

```suggestion
  const classes = classNames(className, children.props.className)
```

Reply:
Yeah I think we just remove `euiScreenReaderOnly` from the component 👍

Reply:
Woot, sounds good!